### PR TITLE
[TACHYON-974] Reenable all test cases in MasterFaultToleranceTest

### PR DESCRIPTION
**Pull request by uronce-cc**
https://tachyon.atlassian.net/browse/TACHYON-974

### CODE REVIEW

**Comment by gpang** [comment]
Jenkins, test this please

File: servers/src/main/java/tachyon/master/journal/JournalWriter.java [hunk 1]
```diff
@@ -143,8 +143,6 @@ public void close() throws IOException {
     if (mEntryOutputStream != null) {
       mEntryOutputStream.close();
     }
-    // Close the ufs.
```
Comment (gpang) [inline servers/src/main/java/tachyon/master/journal/JournalWriter.java]: Please leave a comment here that the ufs is intentionally not closed, and why it is not closed.

**Comment by gpang** [comment]
Jenkins, test this please

**Comment by uronce-cc** [comment]
@gpang I've added comment and rebased commits into one commit, the tests have all passed, PTAL.

**Comment by gpang** [comment]
I have a general question about the cached clients. If we make the LocalTachyonClusterMultiMaster also call:

System.setProperty("fs.hdfs.impl.disable.cache", "true");

would that prevent this issue?

**Comment by uronce-cc** [comment]
@gpang Yes, the cached clients problem can be solved by setting this property, I didn't notice this solution, thanks. 

Another problem for this issue is that when multiple masters are started, they all share the same HDFS cluster, the HDFS cluster will be started the first time a master is created, that is to say, the directories for HDFS name node and data node are set up the first time a master is created, but in original code, when each master is started, the tachyon home will be deleted and recreated again, so in consequent master creation, the directories of HDFS name node and data node are deleted which causes misbehavior for HDFS. 

I've updated the commit.

**Comment by gpang** [comment]
Thanks for this fix, @uronce-cc !

About your comment about the HDFS setup, is there an ideal way to set up the cluster for tests, or is the current change in this PR the best way? (If the ideal way requires refactoring, that should not be in this PR)

**Comment by gpang** [comment]
LGTM

**Comment by uronce-cc** [comment]
@gpang The current way is the quickest and smallest fix I came up, it works fine and the logic I changed is straightforward, so I think it's fine to go with this fix now.

**Pull request merged**